lib/faraday/request.rb

```diff
@@ -26,7 +26,9 @@ module Faraday
   #   @return [RequestOptions] options
   #
   # rubocop:disable Style/StructInheritance
-  class Request < Struct.new(:method, :path, :params, :headers, :body, :options)
+  class Request < Struct.new(
+    :http_method, :path, :params, :headers, :body, :options
```

Comment:
As far as I agree with the renaming, this is a breaking change in case someone is using or extending the `Request` struct, and thus it can't be merged. Was this change triggered for some specific reason?

Reply:
I know that's breaking, but I think the earlier we break this — the better.

The root problem was in RuboCop offense:

```
Lint/StructNewOverride: `:method` member overrides `Struct#method` and it may be unexpected.	
```

We can disable this offense, but I agree that we should not re-define such core methods.

[`Object.method`](https://rubyapi.org/2.7/o/object#method-i-method)

Reply:
Oh I see. Again, it totally makes sense and I'd like to do it as well, but I'm really worried about introducing a breaking change like this in a hotfix/minor release, it could backfire.

@olleolleolle what do you think? I'd add this to the v2 list of fixes and just disable the offence for this struct for the time being?

Reply:
If you want to delay v2 with this change — we also can add a warning about deprecation, and working alias for now.

Reply:
I've done this. Now, we have deprecated (but working) `Request#method`. I see no date for [`v2` milestone](https://github.com/lostisland/faraday/milestone/10), so I've set June 2020 (I can change it if you want).

I'm not sure about `method=`, I see no tests for it.

And there is warning during tests from result check (it suppressing when we check the output itself), and I think it's OK, but I can suppress it if you want (in tests of method result, and leave tests for the output itself).

Reply:
My take: Avoid dates - we just don't know when a release is happening.

Reply:
> My take: Avoid dates - we just don't know when a release is happening.

It's not like "guaranteed", it's like "not earlier than". Do you still want to drop date?

Reply:
Readers will compare to any date given and may make interpretations, so I'm still with "no dates, but version numbers are alright".

Reply:
My bad: I meant "exactly", not "guaranteed", because "not earlier than" is kind of guaranteed too.

---



> Readers will compare to any date given and may make interpretations, so I'm still with "no dates, but version numbers are alright".

I can agree, but these arguments are required: https://ruby-doc.org/stdlib-2.7.1/libdoc/rubygems/rdoc/Gem/Deprecate.html#method-c-deprecate

Do you want me to implement a custom warning?

Reply:
Ah, my whiny-sounds are powered by following the development of the code which is replacing that thing which used to have dates mentioned. A long and good exploration of how to tell users to stop using something which will be going away, definitely, in the future. There's nuance, and compatibility-keeping in those PRs. To walk backwards from the place it all ended at: https://github.com/rubygems/rubygems/issues/3512

So, yes, a small method of our own is probably easiest.

Reply:
Thank you for explanation.

> So, yes, a small method of our own is probably easiest.

Done.